fs/dafsclient/rmtclient.cpp

```diff
@@ -727,13 +727,16 @@ void CRemoteBase::connectSocket(SocketEndpoint &ep, unsigned connectTimeoutMs, u
             //PrintStackReport();
         }
         bool ok = true;
+        unsigned connecttimeoutMs = DEFAULT_CONNECT_TIME;
         try
         {
+            CCycleTimer timer;
             if (tm.timemon)
             {
                 unsigned remaining;
                 if (tm.timemon->timedout(&remaining))
                     THROWJSOCKEXCEPTION(JSOCKERR_connection_failed);
+                connecttimeoutMs = remaining;
```

Comment:
this means there's 2 timing constructs, it would be better/cleaner to unify on 1.

Also I've noticed a bizarre bug in the existing sRFTM.
It should be deleted and replaced. There's only 2 uses, this being 1 of them, and the timemon in it will never be null.

I think this code should either be:
```
            unsigned remainingMs = timer.remainingMs(connectTimeoutMs);
            if (0 == remainingMs)
                THROWJSOCKEXCEPTION(JSOCKERR_timeout_expired);
            socket.setown(ISocket::connect_timeout(ep, remainingMs));
```

OR, and I think preferable given other existing code in this method reliant on CTimeMon:
```
            unsigned remainingMs;
            if (timeMon.timedout(&remainingMs))
                THROWJSOCKEXCEPTION(JSOCKERR_timeout_expired);
            socket.setown(ISocket::connect_timeout(ep, remainingMs));
```
( and replace sRFTM tm(connectTimeoutMs); with CTimeMon timeMon(connectTimeoutMs); )


and similar for the secure_connect below.